pyro/contrib/gp/models/gpr.py

```diff
@@ -43,9 +41,11 @@ def model(self):
         kernel = self.kernel
         noise = self.get_param("noise")
 
-        K = kernel(self.X) + noise.expand(self.num_data).diag()
-        zero_loc = torch.tensor(K.data.new([0])).expand(self.num_data)
-        pyro.sample("y", dist.MultivariateNormal(zero_loc, K), obs=self.y)
+        K = kernel(self.X) + noise.expand(self.X.size(0)).diag()
+        zero_loc = K.new([0]).expand(K.size(0))
+        # correct event_shape for y
+        y = self.y.t() if self.y.dim() == 2 else self.y
+        pyro.sample("y", dist.MultivariateNormal(zero_loc, K), obs=y)
```

Comment:
ditto: you'll need to either enclose in an `iarange` or `.reshape(extra_event_dims=y.dim() - 1)`